beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1685,14 +1685,20 @@ fn check_block_against_finalized_slot<T: BeaconChainTypes>(
     //
     // Fork choice has the most up-to-date view of finalization and there's no point importing a
     // block which conflicts with the fork-choice view of finalization.
-    let finalized_slot = chain
-        .canonical_head
-        .cached_head()
-        .finalized_checkpoint()
+    let finalized_checkpoint = chain.canonical_head.cached_head().finalized_checkpoint();
+    let finalized_slot = finalized_checkpoint
         .epoch
         .start_slot(T::EthSpec::slots_per_epoch());
 
-    if block.slot() <= finalized_slot {
+    let would_revert_finalized_slot = if block.slot() < finalized_slot {
+        true
+    } else if block.slot() == finalized_slot {
+        block_root != finalized_checkpoint.root
+    } else {
+        false
+    };
```

Comment:
I don't think this change actually does anything, because re-importing the finalized block in forwards sync will result in a `DuplicateFullyImported` status and no data being written.

Despite this, when syncing from a checkpoint block on an epoch boundary (not a skip), backfill has us covered:

> Nov 18 14:39:30.480 DEBUG Re-importing historic block                   block_root: 0xdbf8afdfbc3be1c714bcbc451f9569c14bb68e6bd6a85dea820bf6d2758f66cd, slot: 1768544
> Nov 18 14:39:30.481 DEBUG Writing data columns to store                 block_root: 0xdbf8afdfbc3be1c714bcbc451f9569c14bb68e6bd6a85dea820bf6d2758f66cd, count: 64

Tested on Hoodi. And I confirmed the columns for that block are available on the HTTP API.